File: frontend/src/api/explainability.ts
import { post } from './client';
import type { ExplainRequest, ExplanationResult } from '@/types';

export const explainabilityApi = {
  explain: (modelId: string, req: ExplainRequest): Promise<ExplanationResult> =>
    post<ExplanationResult>(`/models/${modelId}/explain`, req),
};

File: backend/routers/explainability.py
"""SHAP and LIME explainability endpoint."""
from __future__ import annotations

from fastapi import APIRouter

from schemas.explainability import ExplainRequest, ExplainResponse
from services.explainability_service import run_explanation

router = APIRouter()


@router.post("/models/{model_id}/explain")
async def explain(model_id: str, req: ExplainRequest):
    result = run_explanation(
        model_id=model_id,
        input_row=req.input_row,
        method=req.method,
        max_display=req.max_display_features,
    )
    return result
